### PR TITLE
std.c: adding freebsd's kinfo_proc type.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: lib/std/c/freebsd.zig [hunk 1]
```diff
@@ -629,6 +630,191 @@ comptime {
...
+pub const thread = opaque {};
+pub const vnode = opaque {};
+pub const filedesc = opaque {};
+pub const pwddesc = opaque {};
+pub const vmspace = opaque {};
+pub const pcb = opaque {};
+pub const lwpid_t = i32;
+pub const fixpt_t = u32;
+pub const vm_size_t = if (builtin.cpu.arch.ptrBitWidth() == 32) u32 else u64;
+pub const segsz_t = if (builtin.cpu.arch.ptrBitWidth() == 32) i32 else i64;
```
Comment (andrewrk) [inline lib/std/c/freebsd.zig:661]:
```suggestion
pub const vm_size_t = usize;
pub const segsz_t = usize;
```

File: lib/std/c/freebsd.zig [hunk 2]
```diff
@@ -629,6 +630,191 @@ comptime {
...
+        pr_size: c_ulong,
+        pr_off: c_ulong,
+        pr_scale: c_ulong,
+    },
+    p_start: timeval,
+};
+
+pub const user = extern struct {
+    u_stats: pstats,
+    u_kproc: kinfo_proc,
```
Comment (andrewrk) [inline lib/std/c/freebsd.zig:682]:
```suggestion
    stats: pstats,
    kproc: kinfo_proc,
```

zig std lib convention is to erase prefixes like this on all fields.

File: lib/std/c/freebsd.zig [hunk 3]
```diff
@@ -629,6 +630,191 @@ comptime {
...
+};
+
+pub const KINFO_PROC_SIZE = switch (builtin.cpu.arch) {
+    .x86 => 768,
+    .arm => 816,
+    else => 1088,
+};
+
+comptime {
+    std.debug.assert(@sizeOf(kinfo_proc) == KINFO_PROC_SIZE);
```
Comment (andrewrk) [inline lib/std/c/freebsd.zig:815]:
```suggestion
    assert(@sizeOf(kinfo_proc) == KINFO_PROC_SIZE);
```

**Review by andrewrk — CHANGES_REQUESTED** [review]

**Comment by devnexen** [comment]
cc @Vexu 🙂

**Pull request merged**